### DISCUSSION TITLE
Any way to make Alpine reevaluate x-show, x-bind etc. after using Morph?

### DISCUSSION THREAD

**Discussion by bojandurmic** [alpinejs/alpine#3731] [1. Help]
Hi all,

I am running into an issue with Alpine Morph plugin. I am using `fetch` to download some part of HTML, and then `Alpine.morph` to merge that with my existing Alpine component. 

The issue is - Alpine directives like `x-show`, `x-bind` etc. are not reevaluated after Morph plugin is done. HTML generated by the server has no way of knowing what the state of my Alpine variables is, so it returns HTML without `style="display: none;"` that was created by Alpine during runtime. Also, any dynamically toggled classes through `:class` won't work.

Is there any way to tell Alpine "Hey, you should re-run x-show, x-bind now, to make sure this HTML is in sync with the current state"?

Any help is greatly appreciated. Thanks!

**Comment by ekwoka** [comment]
There isn't currently...

🤔

**Comment by trych** [comment]
Not sure about your exact setup (and also i haven't used the morph plugin), but depending on your setup it might be possible to set your expression to a reactive (x-)data property and to then re-set this data property value when morph is finished. That way the expression would be re-evaluated, even if it would be the same expression as before.

**Comment by bojandurmic** [comment]
I sort of kind of solved the issue in relatively clean way. The underlying issue I had was the fact that Alpine's state was correct, but the HTML returned by the server contained classes and style attributes and whatnot of the old state. This is what I did:

On elements which I wanted to ignore, but still go through their children and update them, I added a `data-morph-children-only="true"` attribute, for example:

```
<div
  data-morph-children-only="true"
  x-data
  class="z-60 fixed bg-white w-full md:max-w-[480px] right-0 top-0 bottom-0 border-r-black border-opacity-40 overscroll-contain overflow-auto"
  x-show="$store.cartOpen"
>
  My stuff goes here
</div>
```

And on elements which I wanted to ignore entirely, children included, I added a `data-morph-ignore="true"` attribute:

```
<button
  data-morph-ignore="true"
  x-show="!$store.navOpen"
  @click="$store.navOpen = true"
  class="text-15 font-medium leading-24 hover:undeline underline-offset-4"
  type="button"
>
  Menu
</button>
```

After I've done that, I altered my Morph code to look like:

```
const newHeader = await getHeaderSection();
Alpine.morph(document.getElementById('shopify-section-header'), newHeader, {
  updating(el, toEl, childrenOnly, skip) {
    if (el.dataset && el.dataset.morphChildrenOnly === 'true') {
      return childrenOnly();
    } else if (el.dataset && el.dataset.morphIgnore === 'true') {
      return skip();
    }
  },
});
```

This allows me to add these attributes to any problematic elements. My issue is fixed, but it still would be great to have an API which will let me programatically reevaluate `x-show`, `x-bind` etc.

**Reply by ekwoka** [reply]
I've done similar.

For skip, children, and replace to have more granular control
